Back to Lower minSDK = API Level 23 - MM6.0

rawmain opened the pull request:
An additional gradle dependency for jackson-datatype-jsr310 is required anyway in order to lower minSDK again from API 26 (Oreo 8.0) to API 23 (Marshmallow 6.0) - reference issue https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/issues/175#issuecomment-891587370 .

It allows indeed Jackson to recognize Java 8 Date & Time API data types (JSR-310) when desugaring has been set/enabled in app/build.gradle + minSDK=23 & the dgca-verifier-app is running on Android 23<=API<26 devices.

Otherwise, an exception is raised.

------

Other changes :

- Changed zxing:core from 3.3.0 to 3.3.2 (default version for zxing-android-embedded 3.6.0) = working OK with API 23+ devices

-  Replaced with paddingLeft/paddingRight in order to avoid LINT rules override. XML android:paddingHorizontal triggers indeed a LINT error when minSDK <26 = AAPT2 supports desugaring backwards-translation for XML attribute android:paddingHorizontal, but only for layout files, not styles - [IssueTracker Reference](https://issuetracker.google.com/issues/145362255)

- Enabled explicit Desugaring support for project revocation [`0559bdc`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/0559bdcaf03e13b5f3081aed435c69580cd9d889)

- Explicited iterator() in order to avoid lint errors without changing gradle version because of the confusion - _between the Kotlin Iterable.forEach extension function and the Java8 forEach function_ - on debug variants (owing to desugaring) [`882aff8`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/882aff8920957a8cb80223c26cc8b730097d2573)

- Using Collections.sort method to sort arrayList in natural order (ascending) & avoid API issues [`1313066`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/13130669e5096f09a77f1d4e02c597bdbe240ae9)

rawmain committed:
Back to Lower minSDK = API Level 23 - MM6.0

An additional gradle dependency for jackson-datatype-jsr310 is required in order to lower minSDK again from API 26 (Oreo 8.0) to API 23 (Marshmallow 6.0).

It allows indeed Jackson to recognize Java 8 Date & Time API data types (JSR-310) when desugaring has been set/enabled in app/build.gradle + minSDK=23 & the dgca-verifier-app is running on Android 23<=API<26 devices.

Otherwise, an exception is raised.

------

Other changes :

- Changed zxing:core from 3.3.0 to 3.3.2 (default version for zxing-android-embedded 3.6.0) = working OK with API 23+ devices

-  XML android:paddingHorizontal triggers a LINT error when minSDK <26. Replaced with paddingLeft/paddingRight in order to avoid LINT rules override = AAPT2 2.19 could support backwards translation, but it would have required to disable/override LINT checks.

rawmain committed:
Back to Lower minSDK = API Level 23 - MM6.0

An additional gradle dependency for jackson-datatype-jsr310 is required in order to lower minSDK again from API 26 (Oreo 8.0) to API 23 (Marshmallow 6.0).

It allows indeed Jackson to recognize Java 8 Date & Time API data types (JSR-310) when desugaring has been set/enabled in app/build.gradle + minSDK=23 & the dgca-verifier-app is running on Android 23<=API<26 devices.

Otherwise, an exception is raised.

------

Other changes :

- Changed zxing:core from 3.3.0 to 3.3.2 (default version for zxing-android-embedded 3.6.0) = working OK with API 23+ devices

-  XML android:paddingHorizontal triggers a LINT error when minSDK <26. Replaced with paddingLeft/paddingRight in order to avoid LINT rules override = AAPT2 2.19 could support backwards translation, but it would have required to disable/override LINT checks.

rawmain committed:
Create android.yml

rawmain committed:
Create android.yml

rawmain committed:
Merge branch 'main' of https://github.com/rawmain/dgca-verifier-app-android-RM into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Delete android.yml

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Remove Guava Version

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

oleksandrsarapulovgl commented on app/src/main/res/values/styles.xml:
```diff
@@ -74,7 +74,9 @@
         <item name="android:letterSpacing">0</item>
         <item name="android:minHeight">48dp</item>
-        <item name="android:paddingHorizontal">@dimen/default_padding</item>
+        <!-- <item name="android:paddingHorizontal">@dimen/default_padding</item> -->
```
@rawmain why do we need to comment these fields?

rawmain replied:
Remove old unnecessary comments. Thanks.

oleksandrsarapulovgl commented on app/build.gradle:
```diff
@@ -129,6 +130,7 @@ dependencies {
     implementation Deps.log_interceptor
     implementation Deps.gson_converter
     implementation Deps.jackson_kotlin
+    implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.12.0")
```
Could you please use Deps dependency URL here?

rawmain replied:
Done ;). Thanks-

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Applied Reviewed changes

- Removed old unncessary comments

- Aligned jackson.datatype implementation with others (Deps dependencies URL)

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Revocation Desugaring Fix 01

rawmain committed:
iterator Debug Lint Fix 02

Added iterator() in order to avoid the confusion - between the Kotlin Iterable.forEach extension function and the Java8 forEach function -  on debug variants while desugaring

rawmain commented:
March 9th 2022 Additional changes : 

- Enabled explicit Desugaring support for project revocation [`0559bdc`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/0559bdcaf03e13b5f3081aed435c69580cd9d889)

- Explicited iterator() in order to avoid lint errors without changing gradle version because of the confusion - _between the Kotlin Iterable.forEach extension function and the Java8 forEach function_ - on debug variants (owing to desugaring) [`882aff8`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/882aff8920957a8cb80223c26cc8b730097d2573)

rawmain committed:
jackson-datatype-jsr310 Revocation dep Fix 03

Additional gradle dependency for jackson-datatype-jsr310 in revocation project too

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
HashFilter Desugaring Fix 04

- Using Collections.sort method to sort arrayList in natural order

- Replaced hashFilterList.forEach with hashFilterList.iterator().forEach in order to avoid the confusion - between the Kotlin Iterable.forEach extension function and the Java8 forEach function -  on debug variants while desugaring.

rawmain commented:
March 17th 2022 Additional Changes :

- Using Collections.sort method to sort arrayList in natural order (ascending) & avoid API issues [`1313066`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/13130669e5096f09a77f1d4e02c597bdbe240ae9)

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge branch 'eu-digital-green-certificates:main' into main

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Branches Base minSdk 23

Aligned minSdk declaration in library [:branches:base]

rawmain committed:
Desugaring Branches Fix 01

Apply Desugaring Edits to branches\dcc build.gradle

rawmain committed:
Desugaring Branches Fix 02

Apply Desugaring Edits to branches\vc build.gradle

rawmain committed:
iterator Fix 01

Added iterator() in order to avoid the confusion - between the Kotlin Iterable.forEach extension function and the Java8 forEach function -  while desugaring

rawmain committed:
padding fix 01

android:paddingHorizontal requires API level 26 - replaced with paddingLeft/paddingRight (API 23 compatibility)

rawmain committed:
Gradle Scripts Desugaring Alignment

Aligned desugaring edits for app/branches gradle scripts (in order to avoid runtime issues on device with API level < 26)

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
[Test] Increase Gradle daemon memory size

Increase Gradle daemon memory size in order to prevent "GC overhead limit exceeded" exception errors

rawmain commented:
August 30th 2022 Additional Changes after ProGuard updates : 

- Increased Gradle daemon memory size (_up to 6GB_) in order to prevent "GC overhead limit exceeded" exception errors in CI workflow runs

rawmain committed:
Merge remote-tracking branch 'upstream/main'

rawmain committed:
Update AppConfig.kt

rawmain committed:
Revert Java Version

rawmain committed:
Testing Desugaring 2.0.0

rawmain committed:
Update gradle-wrapper.properties

rawmain committed:
Revert to Desugaring 1.1.5

rawmain committed:
Update DefaultProtocolHandler.kt

Explicited iterator() in order to avoid lint errors without changing gradle version because of the confusion - between the Kotlin Iterable.forEach extension function and the Java8 forEach function - on debug variants (owing to desugaring)

rawmain committed:
[Test] Increase Gradle daemon memory size

rawmain committed:
Upgraded Desugaring to 2.0.0

Upgraded Desugaring Libs from 1.1.5 to 2.0.0.

Related dependency/plugin changes :

- Gradle Wrapper 6.5 -> 7.5.1
- AGP (Android Gradle Plugin) 4.1.3 -> 7.4.0-beta05

- Hilt (Dagger) 2.38.1 -> 2.42
- Removed dependency androidx.hilt:hilt-lifecycle-viewmodel - since it's already included in Hilt 2.42

rawmain committed:
[Test] Fix for Github CI Workflow

Edit in order to fix "Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)" with Github Actions

rawmain committed:
Increase MaxMetaspaceSize

rawmain committed:
Arrange GRADLE_OPTS

rawmain committed:
disable ExpiredTargetSdkVersion for decoder

Github Actions with AGP 7.4.0-beta05 incorrectly triggers a LINT error with decoder module.

Such edit allows to bypass this bug.

rawmain committed:
disable ExpiredTargetSdkVersion for decoder

Github Actions with AGP 7.4.0-beta05 incorrectly triggers a LINT error with decoder module.

Such edit on project gradle allows to bypass this bug for all modules

rawmain committed:
DCC Manifest Edit for Zxing

Edit in order to avoid bug with Github Action - Manifest merger failed : uses-sdk:minSdkVersion 23 cannot be smaller than version 24 declared in library [com.journeyapps:zxing-android-embedded:4.2.0]

rawmain committed:
DCC Manifest Edit for Zxing

Edit in order to avoid bug with Github Action - Manifest merger failed : uses-sdk:minSdkVersion 23 cannot be smaller than version 24 declared in library [com.journeyapps:zxing-android-embedded:4.2.0]

rawmain commented:
November 23, 2022 Additional Changes after Dependencies Upgrade :

- **Full Upgrade for Desugaring Libs from 1.1.5 to 2.0.0 to support >=API 23**

> Related dependency/plugin changes :
>
> - Gradle Wrapper 6.5 -> 7.5.1
> - AGP (Android Gradle Plugin) 4.1.3 -> 7.4.0-beta05 (*)
> - Hilt (Dagger) 2.38.1 -> 2.42
> - Removed dependency androidx.hilt:hilt-lifecycle-viewmodel - since it's already included in Hilt 2.42
>
> _(*) AGP 7.4.0-beta05 is supported by Android Studio since release Electric Eel (2022.1.1) Beta 5_

- **Explicited iterator() for DefaultProtocolHandler.kt in order to avoid lint errors without changing gradle version because of the confusion - _between the Kotlin Iterable.forEach extension function and the Java8 forEach function_ - on debug variants (owing to desugaring)** [`8ac1091`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/8ac1091f3f1f46a075da86c76ed37c41e187a6e1)

- **CI edit + GRADLE_OPTS in order to avoid LINT issues with Github Actions** [`448a70f`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/448a70fda9d7f5268f1ce1079ab563755cc7f39a)

- **Override to bypass CI LINT bug by disabling the ExpiredTargetSdkVersion LINT check for all projects/modules** [`6a7ed9d`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/6a7ed9d96239333a455f80977c8532c60f911cd4) 

- **Override Zxing Camera for DCC branch** [`23e67b4`](https://github.com/eu-digital-green-certificates/dgca-verifier-app-android/pull/202/commits/23e67b4c5efa59eae9d9da4280393aeef4dc808a)
